### PR TITLE
docs: bring up downstream changes found in C8 REST API's generated docs

### PR DESCRIPTION
**Pull request by pepopowitz**
## Description

This PR addresses a few small C8 REST API spec issues, most of which were identified in https://github.com/camunda/camunda-docs/pull/4427. There are a few extra changes that were identified from pattern-matching the issues in 4427. 

See https://github.com/camunda/camunda-docs/pull/4427 for original conversation on these changes.

### CODE REVIEW

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 1]
```diff
@@ -460,7 +460,7 @@ paths:
             type: integer
             format: int64
       responses:
-        '200':
+        "200":
```
[the remark this answers is not in the document]
Comment (akeller) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:463]: Quick comment - some of them are "" already in the source, particularly the 400 responses (line 1298+). Is this your code editor, or prettier?

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 2]
```diff
@@ -2744,7 +2742,7 @@ components:
           description: Version tag of this process definition.
           type: string
         processDefinitionId:
-          description:  Process definition id of this process definition.
+          description: Process definition id of this process definition.
```
Comment (mesellings) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:2745]:
```suggestion
          description: Process definition ID of this process definition.
```

**Review by mesellings — APPROVED** [review]
Lgtm 👍 

Only follow-up question/proposal for future would be something around telling people how to resolve an error. E.g., for a 400 error, should we say more, like "Bad request. The Authorization could not be patched. Check that your request and endpoint URL are formed correctly." - either in the error itself, or in an HTTP response codes page in the API docs somewhere? Might be overkill, but I feel it could be a gap for less experienced developers/business users who start playing with the API. Anyway, just a suggestion 👍

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tmetzke) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:463]: Let's try to align on using double quotes in general consistently, this is used in this file much more than the single quotes 👍

**Review by tmetzke — APPROVED** [review]
Great initiative, @pepopowitz, thank you! 👍

**Review by mesellings — APPROVED** [review]
Lgtm 👍

**Pull request merged**